packages/core/src/provider.ts

```diff
@@ -345,9 +347,10 @@ export async function getBalance(
   try {
     const wallet = wallets.find(wallet => wallet.label)
     const provider = wallet.provider
-    const balanceHex = await provider.request({ method: 'eth_getBalance', params:[address,'latest'] });
-    return balanceHex
-      ? { [chain.token || 'eth']: shortenHexToString(balanceHex) }
+    const balanceHex = await provider.request({ method: 'eth_getBalance', params:[address,'latest'] })
+    const balanceWei = new BigNumber(parseInt(balanceHex, 16))
+    return balanceWei
+      ? { [chain.token || 'eth']: weiToEth(balanceWei).slice(0,6) }
```

Comment:
I am still not sure why we need to `slice` here? Was that in the original code? I think we should return the full decimal here and then let it be truncated or rounded where ever it is rendered. We pass this balance to all subscriptions, so dapps will most likely want the full value and then decide what to do with it.